src/main/java/ca/dmoj/java/SubmissionAgent.java

```diff
@@ -22,29 +21,17 @@ public static void premain(String argv, Instrumentation inst) throws Unsupported
 
         final Thread selfThread = Thread.currentThread();
 
+        if (!unsafe) {
+            inst.addTransformer(new DisallowedClassesClassFileTransformer(
+                new DisallowedClassRule("sun/misc/Unsafe", UnsafeDisallowedException::new)
+            ));
+        }
         if (noBigMath) {
-            inst.addTransformer(new ClassFileTransformer() {
-                @Override
-                public byte[] transform(ClassLoader loader, String className, Class<?> classBeingRedefined,
-                                        ProtectionDomain protectionDomain, byte[] classfileBuffer)
-                                        throws IllegalClassFormatException {
-                    if (className == null) return classfileBuffer;
-
-                    RuntimeException disallowed = null;
-                    // If the class ever loaded it's because a submission used it
-                    if (className.startsWith("java/math/BigInteger") ||
-                        className.startsWith("java/math/MutableBigInteger")) {
-                        disallowed = new BigIntegerDisallowedException();
-                    } else if (className.startsWith("java/math/BigDecimal")) {
-                        disallowed = new BigDecimalDisallowedException();
-                    }
-
-                    if (disallowed != null) dumpExceptionAndExit(disallowed);
-
-                    // Don't actually retransform anything
-                    return classfileBuffer;
-                }
-            });
+            inst.addTransformer(new DisallowedClassesClassFileTransformer(
+                new DisallowedClassRule("java/math/BigInteger", BigIntegerDisallowedException::new),
+                new DisallowedClassRule("java/math/MutableBigInteger", BigIntegerDisallowedException::new),
+                new DisallowedClassRule("java/math/BigDecimal", BigDecimalDisallowedException::new)
+            ));
```

Comment:
Wonder if it makes sense to use one transformer and just alter the parameters...